script.module.resolveurl/lib/resolveurl/plugins/wecima.py

```diff
@@ -0,0 +1,48 @@
+"""
+    Plugin for ResolveUrl
+    Copyright (C) 2024 Ghb3245
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+from resolveurl.lib import helpers
+from resolveurl.resolver import ResolveUrl, ResolverError
+from resolveurl import common
+from six.moves import urllib_parse
+
+class WecimaResolver(ResolveUrl):
+    name = 'Wecima'
+    domains = ['wecima.show']
+    pattern = r'(?://|\.)(wecima\.show)/run/([^\n]+)'
+
+    def get_media_url(self, host, media_id):
+        if '$$' in media_id:
+            media_id, referer = media_id.split('$$')
+            referer = urllib_parse.urljoin(referer, '/')
+        else:
+            referer = False
+        web_url = self.get_url(host, media_id)
+        if not referer:
+            referer = urllib_parse.urljoin(web_url, '/')
+        headers = {'User-Agent': common.FF_USER_AGENT,
+                   'Referer': referer,
+                   'X-Requested-With': 'XMLHttpRequest'}
+        html = self.net.http_GET(web_url, headers=headers).content
+        sources = re.search(r'source\s*src="(?P<url>[^"]+)', html).group(1)
```

Comment:
This will break as the whole purpose of doing a search is to see if there is a match, and blindly calling group(1) without check will cause `None has no group` error message
So this line should be `sources = re.search(r'source\s*src="(?P<url>[^"]+)', html)`